### DISCUSSION TITLE
Adventure on Win7

### DISCUSSION THREAD

**Discussion by brynne8** [nuejs/nue#178] [Show and tell]
I recently embarked on a journey to use Nue.js on Windows 7 and I'm excited to share that it's successfully running, albeit with some deviations from the documentation.

As a Windows 7 user, I powered Nue.js with Node.js (specifically version 18). The first obstacle I encountered was that much of the documentation seemed tailored for the Bun runtime, rather than Node. For instance, the documentation suggests using the command:

```
node $(which nue)
```

However, this doesn't work on Windows as the `which` command is unavailable. To overcome this, I modified the `nue.cmd` file, replacing `bun` with `node`. This allowed me to run the `nue` command, but I was then faced with another issue:

```
Bundler not found. Please use Bun or install esbuild
```

Despite having already added `esbuild` via `yarn add esbuild` (I prefer local installations over global ones), Nue.js didn't seem to recognize it. The solution was to install `esbuild` globally, which resolved this issue.

My next challenge was in creating a simple Single Page Application (SPA) by following the documentation. The [Create a Skeleton App](https://nuejs.org/docs/tutorials/build-a-simple-spa.html#create-a-skeleton-app) section was particularly confusing. In tutorials, I like to follow steps sequentially, but I found myself grappling with `<nue-island island="app"></nue-island>` and then `<app :title :description/>`, without prior instructions on configuring `site.yaml` or understanding the rendering process. This section, I believe, would benefit from a more step-by-step approach.

I wanted to share my experience in case others are facing similar issues and to potentially discuss improvements to the documentation for better clarity, especially for those of us using Windows environments.

**Comment by tipiirai** [comment]
Thank you for this @brynne8! Great feedback. This helps me improve the documentation. I know I rushed that SPA section since it's not currently a priority as you can see from [our plan](https://nuejs.org/blog/perfect-web-framework/).  And of course, need to fix the esbuild resolve issue on local installations. Could be Windows specific, but likely not.
